nemo_skills/evaluation/metrics.py

```diff
@@ -731,17 +744,36 @@ def compute_metrics(
     max_samples=-1,
     aggregation_mode='first',
 ):
-    metrics_calculator.reset()
+
     metrics_calculator.setup(input_files)
+    metrics_calculator.reset()
 
     file_handles = [open(file, "rt", encoding="utf-8") for file in unroll_files(input_files)]
-    for idx, predictions in enumerate(zip_longest(*file_handles)):
-        if idx == max_samples:
-            break
-        data = read_predictions(predictions, metrics_calculator, allow_incomplete)
-        metrics_calculator.update(data, aggregation_mode)
+
+    ### default metrics include majority vote and pass@k
+    aggregation_mode_dict = {"majority": "majority", "best": "pass"}
```

Comment:
Instead of doing it here, I suggest that you directly do it in each class. So here the code would stay mostly as it was before, except that we don't take aggregation mode as a parameter. But each class would return instead of a single dictionary with metrics, a dictionary of dictionaries, which would be `{'greedy': {greedy metrics}, 'majority': {majority metrics}}` and so on depending on what makes most sense for that particular eval